roles/grafana/templates/datasources.yml.j2

```diff
@@ -0,0 +1,28 @@
+apiVersion: 1
+datasources:
+
+{% if influxdb_flux_query %}
+  - name: InfluxDB_v2_Flux
```

Comment:
I think it would be nice to make the name of the data source configurable, i.e. with a variable named `grafana_influxdb_datasource_name`, same applies for line 18 of this template.

Reply:
After talking with @Saeid-Abadi again we came up with the following idea for generating an arbitrary amount of different data sources from a single nested variable to be placed in the `default/main.yml` file. @widhalmt what do you think of it?

```yaml
grafana_datasources:
  icingaweb2_influxql:
    type: influxdb
    access: proxy
    url: "{{ grafana_influxdb_url }}"
    database: "{{ grafana_influxdb_bucket_name }}"
    jsonData:
      httpMode: GET
      httpHeaderName1: 'Authorization'
    secureJsonData:
      httpHeaderValue1: 'Token {{ influxdb_graphing_token }}'
  icingaweb2_flux:
    type: influxdb
    access: proxy
    url: http://prometheus.influx.cluster.com
    database: prometheus-db
    jsonData:
      version: Flux
      organization: "{{ grafana_influxdb_organization }}"
      bucket: "{{ grafana_influxdb_bucket_name }}"
      tlsSkipVerify: true
  other_data_mysql:
    type: MySQL
    access: proxy
    url: http://randomdata.mysql.otherserver.com
    database: backup-db
    user: db-user
    password: db-password
```

This could get extended with any other datasource type, address, etc., either with variables as data or with hardcoded values, depending on the environment the role is used in. I added a fictional MySQL database (it's not correctly configured and just servers as an example) to the example. The whole representation could get parsed into the needed `.ini` format using this macro from the icinga2-role in one go:

https://github.com/Saeid-Abadi/ansible-collection-icinga/blob/add-icingaweb2-role/roles/icingaweb2/templates/ini_template.j2

Reply:
Having an option to add arbitrary sources would be great! I'm totally hooked on the idea because it would definitely improve the value of the overall role.

Two things to consider:
* There needs to be at least one extensive example in `README.md` or `docs/` explaining how to use this. Even better, examples for quickstart to use it together with other roles. (`influxdb` now, `graphite` later).
* I'm not abolutely sure that the macro can work on this kind of configuration. Wouldn't be the `jsonData` section be one section too many? Or did I misinterpret the macro?

Reply:
Good spot, right now the macro would not parse the nested structure down to its leaves, we'd need to expand the logic to move through all levels greedily or take a good look at Grafana's config and rewrite the macro to match the expected/possible structure of datasource configurations, e.g. by adding a third level.

Writing proper examples will definitely be necessary but shouldn't be too much of a hassle.

Reply:
I can imagine we don't need more than a third level. Collecting some example configuration shouldn't be very hard, we agree on that, too.